wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/SourceStorageImpl.java

```diff
@@ -64,6 +68,20 @@ public SourceStorageImpl(String type, String location, Map<String, String> param
     }
   }
 
+  @PrePersist
+  @PreUpdate
+  public void validate() {
```

Comment:
It's nice that you added this check and Che Server is safer, but we have a lot of other places where `Map<String, String>` is stored and the same check is not done. Could you please create an issue to fix all other places? Maybe just copy/paste your code or find a common solution.

Reply:
https://github.com/eclipse/che/issues/13081